src/python/tests/core/crash_analysis/stack_parsing/stack_analyzer_test.py

```diff
@@ -2435,3 +2435,174 @@ def test_log_fatal_google(self):
+    data = self._read_test_data('golang_libfuzzer_panic.txt')
+    expected_type = 'ASSERT'
+    expected_address = ''
+    expected_state = (
+        'parse //%B9%B9%B9%B9%B9%01%00%00%00%00%00%00%00%B9%B9%B9%B9%B9%B9%B9%B'
+        '9%B9%B9%B9\nurl.Fuzz\n')
+
+    expected_stacktrace = data
+    expected_security_flag = True
+    self._validate_get_crash_data(data, expected_type, expected_address,
+                                  expected_state, expected_stacktrace,
+                                  expected_security_flag)
+
+  def test_golang_panic_with_type_assertions_in_frames(self):
+    """Test golang stacktrace with panic with type assertions in stack frames.
+    """
+    data = self._read_test_data(
+        'golang_panic_with_type_assertions_in_frames.txt')
+    expected_type = 'ASSERT'
+    expected_address = ''
+    expected_state = ('index > windowEnd\n'
+                      'flate.(*compressor).deflate\n'
+                      'flate.(*compressor).syncFlush\n')
+
+    expected_stacktrace = data
+    expected_security_flag = True
+    self._validate_get_crash_data(data, expected_type, expected_address,
+                                  expected_state, expected_stacktrace,
+                                  expected_security_flag)
+
+  def test_golang_fatal_error_stack_overflow(self):
+    """Test golang stacktrace with fatal error caused by stack overflow."""
+    data = self._read_test_data('golang_fatal_error_stack_overflow.txt')
+    expected_type = 'stack overflow'
+    expected_address = ''
+    expected_state = ('ast.(*scanner).next\n'
+                      'ast.(*scanner).scanIdent\n'
+                      'ast.(*scanner).Scan\n')
+
+    expected_stacktrace = data
+    expected_security_flag = False
+    self._validate_get_crash_data(data, expected_type, expected_address,
+                                  expected_state, expected_stacktrace,
+                                  expected_security_flag)
+
+  def test_golang_panic_custom_short_message(self):
+    """Test golang stacktrace with panic and custom short message."""
+    data = self._read_test_data('golang_panic_custom_short_message.txt')
+    expected_type = 'ASSERT'
+    expected_address = ''
+    expected_state = 'bad hex char\nprog.fromHexChar\nprog.hexToByte\n'
+
+    expected_stacktrace = data
+    expected_security_flag = True
+    self._validate_get_crash_data(data, expected_type, expected_address,
+                                  expected_state, expected_stacktrace,
+                                  expected_security_flag)
+
+  def test_golang_panic_runtime_error_invalid_memory_address(self):
+    """Test golang stacktrace with panic caused by invalid memory address."""
+    data = self._read_test_data(
+        'golang_panic_runtime_error_invalid_memory_address.txt')
+    expected_type = 'invalid memory address'
+    expected_address = ''
+    expected_state = (
+        'runtime error: invalid memory address or nil pointer dereference\n'
```

Comment:
yes this frame shouldnt be in crash state, see comment above.

Reply:
Done